test/cache.js

```diff
@@ -34,7 +34,14 @@ test('cache skipped when the GQL Schema has been changed', async t => {
   const app = Fastify()
   t.teardown(() => app.close())
 
-  await app.register(GQL, { schema, resolvers, jit: 1 })
+  await app.register(GQL, {
+    schema,
+    resolvers,
+    jit: 1,
+    compilerOptions: {
+      customJSONSerializer: true
+    }
```

Comment:
This test is not verifying anything. You'd need to use proxyquire to verify you are passing the parameters correctly.

Reply:
I have added a new test which checks if we are passing the `compilerOptions` correctly to the function. Please let me know if the test needs any changes.